File: tests/human/x/scala/cross_join.scala
object cross_join {
  case class Customer(id: Int, name: String)
  case class Order(id: Int, customerId: Int, total: Int)
  case class Pair(orderId: Int, orderCustomerId: Int, pairedCustomerName: String, orderTotal: Int)

  def main(args: Array[String]): Unit = {
    val customers = List(
      Customer(1, "Alice"),
      Customer(2, "Bob"),
      Customer(3, "Charlie")
    )
    val orders = List(
      Order(100, 1, 250),
      Order(101, 2, 125),
      Order(102, 1, 300)
    )

    val result = for {
      o <- orders
      c <- customers
    } yield Pair(o.id, o.customerId, c.name, o.total)

    println("--- Cross Join: All order-customer pairs ---")
    result.foreach { entry =>
      println(s"Order ${entry.orderId} (customerId: ${entry.orderCustomerId}, total: $${entry.orderTotal}) paired with ${entry.pairedCustomerName}")
    }
  }
}

File: tests/human/x/cpp/cross_join.cpp
#include <iostream>
#include <vector>
#include <string>
struct Customer {int id; std::string name;};
struct Order {int id; int customerId; int total;};
int main(){
    std::vector<Customer> customers={{1,"Alice"},{2,"Bob"},{3,"Charlie"}};
    std::vector<Order> orders={{100,1,250},{101,2,125},{102,1,300}};
    std::cout << "--- Cross Join: All order-customer pairs ---\n";
    for(const auto &o:orders){
        for(const auto &c:customers){
            std::cout << "Order "<<o.id
                      << "(customerId:"<<o.customerId
                      << ", total:$"<<o.total
                      << ") paired with "<<c.name<<"\n";
        }
    }
    return 0;
}
